src/plugins/visualizations/public/utils/saved_visualize_utils.ts

```diff
@@ -0,0 +1,392 @@
+  const searchOption = (field: string, ...defaults: string[]) =>
+    _(extensions).map(field).concat(defaults).compact().flatten().uniq().value() as string[];
+  const searchOptions: SavedObjectsFindOptions = {
+    type: searchOption('docTypes', 'visualization'),
+    searchFields: searchOption('searchFields', 'title^3', 'description'),
+    search: search ? `${search}*` : undefined,
+    perPage: size,
+    page: 1,
+    defaultSearchOperator: 'AND' as 'AND',
+    hasReference: references,
+  };
+
+  const { total, savedObjects } = await savedObjectsClient.find<SavedObjectAttributes>(
+    searchOptions
+  );
+
+  return {
+    total,
+    hits: savedObjects.map((savedObject) => {
+      const config = extensionByType[savedObject.type];
+
+      if (config) {
+        return {
+          ...config.toListItem(savedObject),
+          references: savedObject.references,
+        };
+      } else {
+        return mapHitSource(visTypes, savedObject);
+      }
+    }),
+  };
+}
+
+export async function getSavedVisualization(
+  services: {
+    savedObjectsClient: SavedObjectsClientContract;
+    search: DataPublicPluginStart['search'];
+    dataViews: DataPublicPluginStart['dataViews'];
+    spaces: SpacesPluginStart;
+    savedObjectsTagging?: SavedObjectsTaggingApi;
+  },
+  opts?: any
+): Promise<VisSavedObject> {
+  if (typeof opts !== 'object') {
+    opts = { id: opts };
+  }
+
+  const id = (opts.id as string) || '';
+  const savedObject = {
+    id,
+    copyOnSave: false,
+    migrationVersion: opts.migrationVersion,
+    displayName: SAVED_VIS_TYPE,
+    getEsType: () => SAVED_VIS_TYPE,
+    getDisplayName: () => SAVED_VIS_TYPE,
+    searchSource: opts.searchSource ? services.search.searchSource.createEmpty() : undefined,
+  } as VisSavedObject & { _source: any };
+  const config = { injectReferences };
+
+  const defaultsProps = getDefaults(opts);
+
+  if (!id) {
+    Object.assign(savedObject, defaultsProps);
+    return savedObject;
+  }
+
+  const {
+    saved_object: resp,
+    outcome,
+    alias_target_id: aliasTargetId,
+  } = await services.savedObjectsClient.resolve<SavedObjectAttributes>(SAVED_VIS_TYPE, id);
+
+  if (!resp._version) {
+    throw new SavedObjectNotFound(SAVED_VIS_TYPE, id || '');
+  }
+
+  savedObject._source = _.cloneDeep(resp.attributes);
+
+  if (savedObject._source.visState) {
+    savedObject._source.visState = JSON.parse(savedObject._source.visState);
+  }
+
+  // assign the defaults to the response
+  _.defaults(savedObject._source, defaultsProps);
+
+  Object.assign(savedObject, savedObject._source);
+  savedObject.lastSavedTitle = savedObject.title;
+
+  savedObject.sharingSavedObjectProps = {
+    aliasTargetId,
+    outcome,
+    errorJSON:
+      outcome === 'conflict' && services.spaces
+        ? JSON.stringify({
+            targetType: SAVED_VIS_TYPE,
+            sourceId: id,
+            targetSpace: (await services.spaces.getActiveSpace()).id,
```

Comment:
ait services.spaces?.getActiveSpace()).id,

Reply:
no needed, above I check if we have spaces